mapie/estimators.py

```diff
@@ -96,85 +98,109 @@ class MapieRegressor(BaseEstimator, RegressorMixin):  # type: ignore
     >>> from sklearn.linear_model import LinearRegression
     >>> X_toy = np.array([0, 1, 2, 3, 4, 5]).reshape(-1, 1)
     >>> y_toy = np.array([5, 7.5, 9.5, 10.5, 12.5, 15])
-    >>> pireg = MapieRegressor(LinearRegression(), method="jackknife_plus")
+    >>> pireg = MapieRegressor(LinearRegression())
     >>> print(pireg.fit(X_toy, y_toy).predict(X_toy))
-    [[ 5.28571429  4.61627907  6.2       ]
-     [ 7.17142857  6.51744186  8.        ]
-     [ 9.05714286  8.4         9.8       ]
-     [10.94285714 10.2        11.6       ]
+    [[ 5.28571429  4.61627907  6.        ]
+     [ 7.17142857  6.51744186  7.8       ]
+     [ 9.05714286  8.4         9.68023256]
+     [10.94285714 10.2        11.58139535]
      [12.82857143 12.         13.48255814]
      [14.71428571 13.8        15.38372093]]
     """
 
-    valid_methods = [
+    valid_methods_ = [
         "naive",
-        "jackknife",
-        "jackknife_plus",
-        "jackknife_minmax",
-        "cv",
-        "cv_plus",
-        "cv_minmax"
+        "base",
+        "plus",
+        "minmax"
     ]
 
     def __init__(
         self,
         estimator: Optional[RegressorMixin] = None,
         alpha: float = 0.1,
-        method: str = "cv_plus",
-        n_splits: int = 5,
-        shuffle: bool = True,
-        ensemble: bool = False,
-        random_state: Optional[int] = None
+        method: str = "plus",
+        cv: Optional[Union[int, BaseCrossValidator]] = None,
+        ensemble: bool = False
     ) -> None:
         self.estimator = estimator
         self.alpha = alpha
         self.method = method
-        self.n_splits = n_splits
-        self.shuffle = shuffle
+        self.cv = cv
         self.ensemble = ensemble
-        self.random_state = random_state
 
     def _check_parameters(self) -> None:
         """
-        Perform several checks on input parameters and estimator.
+        Perform several checks on input parameters.
         """
-        if not 0 < self.alpha < 1:
-            raise ValueError("Invalid alpha. Please choose an alpha value between 0 and 1.")
-        if self.method not in self.valid_methods:
-            raise ValueError("Invalid method.")
+        if not isinstance(self.alpha, float) or not 0 < self.alpha < 1:
```

Comment:
La condition `not isinstance(self.alpha, float)` est-elle utile ? Peut-on définir une variable entre 0 et 1 qui n'est pas un `float` ?

Reply:
Je ne l'avais pas mise au départ, mais si `alpha` est une string, la condition `alpha < 1` crash. Donc pour que la condition puisse retourner True/False sans erreur, il faut d'abord s'assurer que `alpha est un nombre.

Reply:
OK entendu.